driver-core/src/test/java/com/datastax/driver/core/EventDebouncerIntegrationTest.java

```diff
@@ -0,0 +1,135 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import java.util.Collection;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.TimeUnit;
+
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Fail.fail;
+
+import static com.datastax.driver.core.CCMBridge.ipOfNode;
+
+public class EventDebouncerIntegrationTest {
+
+    /**
+     * Tests that DOWN, UP, REMOVE or ADD events will not be delivered to
+     * load balancing policy nor host state listeners
+     * before the cluster is fully initialized.
+     *
+     * @throws InterruptedException
+     * @jira_ticket JAVA-784
+     * @since 2.0.11
+     */
+    @Test(groups = "short")
```

Comment:
This should probably be marked 'long' since it doesn't use PerClassSingleNodeCluster/takes node down/etc, although it runs decently quickly (~25 secs in my environment).